kubevirt-vmware/pkg/controller/ovirtprovider/client.go

```diff
@@ -24,6 +25,11 @@ type VM struct {
 
 // NewClient creates new client struct based on connection details provided
 func NewClient(ctx context.Context, url string, username string, password string, ca string) (*Client, error) {
+	if !strings.HasPrefix(url, "http") {
```

Comment:
Should we also check for 443, and 8443 ports as discussed in BZ?

Reply:
We expect to work with https. It would be good to provide early feedback to the user.

Reply:
No matter how unlikely it is, you can easily match a hostname starting with `http` and assume it is OK. I would suggest being explicit and check for `http://` and `https://'. But that is just my humble opinion, it is up to you really.

Reply:
Yeah, I wondered what's wrong with parsing the url (https://gobyexample.com/url-parsing) but I guess all the types used in go-ovirt are strings and numbers for some reason.

Reply:
We expect to have protocol as part of url so if not provided we will add it.

Reply:
We understand that. The point was that the check is not foolproof. It will miss-detect when you get string like `http.testenv.dev/foo/bar`. What @nertpinx suggested is that using `url.Parse()` may be better. But now that I had a look on the documentation, it is not very helpful either:

```
Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities. 
```

Reply:
Good point. Pushing an update.